text/0014-native-currency.md

```diff
@@ -0,0 +1,94 @@
+# Native Currency of the Casper Network
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0014](https://github.com/casperlabs/ceps/pull/0014)
+
+The Casper Network has a native currency, called Casper, and denoted by the ticker symbol CSPR.
+
+## Motivation
+
+[motivation]: #motivation
+
+To create a reference for CSPR and how it is denominated.
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+The Casper Network has a native currency, called Casper, and denoted by the ticker symbol CSPR. It is primarily used to pay for transaction fees, but can also serve many other purposes, such as a store of value, collateral, or any function users see fit.
+
+The smallest subdenomination of CSPR is called *mote*. We set 1 CSPR = 10^9 mote. We denote greater subdenominations with SI prefixes:
+
+| Multiplier | Name |
+|-|-|
+| 10^0 | mote |
+| 10^3 | kilomote |
+| 10^6 | megamote |
+| 10^9 | gigamote = Casper |
+
+We represent all Casper related quantities in terms of motes at the implementation level. The number of decimals is chosen so that we will never need to subdivide the smallest subdenomination.
+
+The initial supply of Casper at mainnet launch will be 10 billion. The issuance schedule is out of the scope of this proposal.
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+N/A.
+
+## Drawbacks
+
+[drawbacks]: #drawbacks
+
+N/A.
+
+## Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+We have two goals when we set how Casper is denominated:
+
+- The circulating supply of the base unit, i.e. mote, must be large enough so that we will never need to subdivide it.
+- The accumulation of round-off errors is minimized.
+
+We tried to reason according to these goals for our decision, and made comparisons with other blockchains.
+
+## Prior art
+
+[prior-art]: #prior-art
+
+We define `total_supply` as the total supply of a blockchain's native currency, and `decimals` as the number of decimal places representing its smallest subdenomination.
+
+Then supply of the base unit is computed as `base_supply = total_supply * 10^decimals`.
+
+| Blockchain | Total supply | Decimals | Base supply |
+|-|-|-|-|
+| Bitcoin | 21 million | 8 | ~10^15.32 |
+| Ethereum | ~100 million | 18 | ~10^26 |
+| Polkadot | ~1 billion | 10 | ~10^19 |
```

Comment:
Does Polkadot use 64-bit numbers for amounts? It looks like that's the intention, because
```python3
>>> 2**64 / 10**19
1.8446744073709551
```
With a 2% increase per year, this means our total supply would fit into a `u64` for 30 years:
```python3
>>> log(2**64 / 10**19) / log(1.02)
30.920280414900077
```
Each zero we remove would add about 116 years to that. But I'm comfortable with having our very own "year 2050 problem" and planning for a network upgrade within the next 30 years, too. :+1: 

If we want to use a `u128` from the beginning, however, it feels like we'd be wasting half of the bits. If we do that we might as well add more zeros and smaller rounding errors.

Reply:
Polkadot uses 128-bit too. It looks like it's the norm

https://github.com/paritytech/polkadot/blob/96af6eadd5649e7faef75ddfe3cc181bd1567155/core-primitives/src/lib.rs#L63